// @dasha.ai/sdk/dist/log.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.disableSentryTransport = exports.enableSentryTransport = exports.logAndReturn = exports.logAndThrow = exports.log = void 0;
const winston = require("winston");
const winston_transport_sentry_node_1 = require("winston-transport-sentry-node");
/**
 * A global logger instance.
 *
 * All logger instances are [winston] loggers.
 *
 * [winston]: https://npmjs.com/winston
*/
exports.log = winston.createLogger({
    level: process.env.DASHA_LOG_LEVEL || "info",
});
// https://github.com/winstonjs/logform/issues/100#issuecomment-744318235
exports.log.format = winston.format.errors({ stack: exports.log.isDebugEnabled() });
exports.log.add(new winston.transports.Console({
    format: winston.format.combine(winston.format.colorize(), winston.format.timestamp(), winston.format.printf(({ timestamp, label, level, message, stack }) => `${timestamp} [${label !== null && label !== void 0 ? label : "sdk"}] ${level} ${stack !== null && stack !== void 0 ? stack : message}`)),
}));
function logAndThrow(log, error) {
    log.error(error);
    throw error;
}
exports.logAndThrow = logAndThrow;
function logAndReturn(log, error) {
    log.error(error);
    return error;
}
exports.logAndReturn = logAndReturn;
function enableSentryTransport() {
    if (exports.log.transports.some((t) => t instanceof winston_transport_sentry_node_1.default)) {
        return;
    }
    exports.log.add(new winston_transport_sentry_node_1.default({
        level: "error",
        sentry: {
            dsn: "https://d59ab8bd2df644049e9c0520fc79ca91@sentry.dev.az.dasha.ai/28",
        },
    }));
    exports.log.info("error reporting is enabled", { label: "sdk" });
}
exports.enableSentryTransport = enableSentryTransport;
function disableSentryTransport() {
    const transport = exports.log.transports.find((t) => t instanceof winston_transport_sentry_node_1.default);
    if (transport === undefined)
        return;
    exports.log.remove(transport);
    exports.log.info("error reporting is disabled", { label: "sdk" });
}
exports.disableSentryTransport = disableSentryTransport;
