// @dasha.ai/sdk/dist/queue.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueuedConversation = exports.ConversationQueue = void 0;
const EventEmitter = require("eventemitter3");
const errors = require("./errors");
const sip = require("./sip");
const log_1 = require("./log");
const transcription_1 = require("./transcription");
/**
 * Manages planned conversations.
 *
 * A conversation may be _enqueued_, that is, planned to execute in a specific time frame.
 * When the conversation is to be executed, a `"ready"` event is fired on this object,
 * that gets passed a {@link Conversation} object to populate its input and run.
 *
 * The queue is stored on the Dasha platform. To associate an enqueued conversation
 * with its data, each conversation is assigned a _key_.
 *
 * @example running enqueued conversations
 * ```typescript
 * application.queue.on("ready", async (key, conversation) => {
 *   conversation.input = getInput(key);
 *   const result = await conversation.execute();
 * });
 * ```
 *
 * Sometimes, the conversation can be removed from the queue by the Dasha platform itself.
 * In that case, a `"rejected"` event is fired.
 *
 * @example conversation rejection
 * ```typescript
 * application.queue.on("rejected", (key, error) => {
 *   console.log(`conversation with key ${key} was rejected`, error);
 * });
 * ```
 *
 * There is an important caveat with the incoming SIP conversations.
 * They get placed in the queue automatically, by the Dasha platform itself.
 * To discern them from the manually-enqueued conversations, and to get the additional
 * SIP info (e.g. the SIP user and domain of the conversation partner),
 * the `"ready"` event has an extra argument of type {@link QueuedConversationInfo}.
 *
 * @example using SIP data in a conversation
 * ```typescript
 * application.queue.on("ready", (key, conversation, info) => {
 *   assert(info.sip !== undefined);
 *   conversation.input.from = `${info.fromUser}@${info.fromDomain}`;
 *   await conversation.execute();
 * });
 * ```
 */
class ConversationQueue extends EventEmitter {
    /** @internal */
    constructor(application, log) {
        super();
        this._ignoredJobKeys = new Set();
        this._activeConversationsByJobKey = new Map();
        this._isDisposed = false;
        this._handleJobTimedOut = (_jobId, jobKey) => {
            if (this._ignoredJobKeys.has(jobKey)) {
                this._ignoredJobKeys.delete(jobKey);
                return;
            }
            const conv = this._activeConversationsByJobKey.get(jobKey);
            if (conv !== undefined) {
                conv._timedOut = true;
                this._activeConversationsByJobKey.delete(jobKey);
            }
            this.emit("timeout", jobKey);
        };
        this._handleJobRejected = (_jobId, jobKey, data) => {
            if (this._ignoredJobKeys.has(jobKey)) {
                this._ignoredJobKeys.delete(jobKey);
                return;
            }
            const conv = this._activeConversationsByJobKey.get(jobKey);
            if (conv !== undefined) {
                conv._rejected = true;
                conv._rejectedReason = new errors.ConversationRejectedError(data);
                this._activeConversationsByJobKey.delete(jobKey);
            }
            this.emit("rejected", jobKey, log_1.logAndReturn(this._log, new errors.ConversationRejectedError(data)));
        };
        this._handleJobStarting = (jobId, jobKey, jobData) => __awaiter(this, void 0, void 0, function* () {
            if (this._ignoredJobKeys.has(jobKey)) {
                this._ignoredJobKeys.delete(jobKey);
                return;
            }
            const conv = new QueuedConversation(this._application, jobId, jobKey, this._log);
            const sipInfo = sip._extractSipInfo(jobData);
            this.emit("ready", jobKey, conv, { sip: sipInfo });
            setTimeout(() => {
                if (conv._started)
                    return;
                conv._timedOut = true;
                this._application._rejectJob(jobId).catch((error) => this.emit("error", error));
            }, 2 * 60 * 1000);
        });
        this._application = application;
        this._log = log.child({});
        application.on("_jobTimedOut", this._handleJobTimedOut);
        application.on("_jobRejected", this._handleJobRejected);
        application.on("_jobStarting", this._handleJobStarting);
    }
    /** @internal */
    _dispose() {
        if (this._isDisposed)
            return;
        this._isDisposed = true;
        this.removeAllListeners();
        this._application.off("_jobTimedOut", this._handleJobTimedOut);
        this._application.off("_jobRejected", this._handleJobRejected);
        this._application.off("_jobStarting", this._handleJobStarting);
    }
    /**
     * Push a new conversation into the queue.
     *
     * @param key a conversation key that can be used to associate data
     * @param options.after the earliest point in time the conversation can run at
     * @param options.before the latest point in time the conversation can run at
     *
     * @example enqueue a conversation to within an hour
     * ```typescript
     * application.queue.push("key", {
     *   after: new Date(),
     *   before: new Date(Date.now() + 60 * 60 * 1000)
     * });
     * ```
     */
    push(key, options) {
        var _a, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const after = (_a = options === null || options === void 0 ? void 0 : options.after) !== null && _a !== void 0 ? _a : new Date();
            const before = (_b = options === null || options === void 0 ? void 0 : options.before) !== null && _b !== void 0 ? _b : new Date(after.valueOf() + 5 * 60 * 1000);
            const priority = (_c = options === null || options === void 0 ? void 0 : options.priority) !== null && _c !== void 0 ? _c : 1;
            yield this._application._enqueueJob(key, { after, before, priority });
            this._log.info(`conversation enqueued: ${key}`, { jobKey: key });
        });
    }
    /** @internal */
    _ignore(jobKey) {
        this._ignoredJobKeys.add(jobKey);
    }
}
exports.ConversationQueue = ConversationQueue;
/** @internal */
class QueuedConversation extends EventEmitter {
    constructor(application, jobId, jobKey, log) {
        super();
        this.input = {};
        /** @internal */ this._started = false;
        /** @internal */ this._timedOut = false;
        /** @internal */ this._rejected = false;
        this._transcription = [];
        this._transcriptionParser = new transcription_1.TranscriptionParser();
        this._application = application;
        this._jobId = jobId;
        this._jobKey = jobKey;
        this.log = log.child({
            label: `conv:${this._jobId.slice(-6)}`,
            jobId,
            jobKey,
        });
        application._bindConversation(jobId, this);
        this._transcriptionParser.on("transcription", (transcription) => {
            this.emit("transcription", transcription);
            this._transcription.push(transcription);
        });
    }
    execute(options) {
        var _a, _b;
        return __awaiter(this, void 0, void 0, function* () {
            if (this._timedOut) {
                log_1.logAndThrow(this.log, new errors.TimeoutError("conversation timed out"));
            }
            if (this._rejected) {
                log_1.logAndThrow(this.log, this._rejectedReason);
            }
            if (this._started) {
                log_1.logAndThrow(this.log, new errors.Error("conversation has already been started"));
            }
            this._started = true;
            this.log.info("conversation started");
            const connection = yield this._application._executeConnectionProvider(this);
            connection._sessionConfig.tts = yield this._application._executeTtsDispatcher(this);
            connection._sessionConfig.stt = yield this._application._executeSttDispatcher(this);
            const { configName } = yield this._application._addSessionConfig(connection._sessionConfig);
            if ((_a = options === null || options === void 0 ? void 0 : options.cancelToken) === null || _a === void 0 ? void 0 : _a.isCanceled) {
                yield this._application._rejectJob(this._jobId);
                options.cancelToken._throwIfCanceled(this.log);
            }
            connection._init(this._jobId);
            yield this._application._acceptJob(this._jobId, {
                sessionConfigName: configName,
                data: this.input,
            });
            (_b = options === null || options === void 0 ? void 0 : options.cancelToken) === null || _b === void 0 ? void 0 : _b._throwIfCanceled(this.log);
            return new Promise((resolve, reject) => {
                var _a;
                (_a = options === null || options === void 0 ? void 0 : options.cancelToken) === null || _a === void 0 ? void 0 : _a._onCancel(this.log, reject);
                const unsubscribe = () => {
                    this._application.off("_jobFailed", handleJobFailed);
                    this._application.off("_jobCompleted", handleJobCompleted);
                    this._application.off("_jobRecordIdReady", handleJobRecordIdReady);
                    this._application.off("_jobEvent", handleJobEvent);
                };
                const handleJobFailed = (jobId, data) => {
                    if (jobId !== this._jobId)
                        return;
                    unsubscribe();
                    reject(log_1.logAndReturn(this.log, new errors.ConversationError(data)));
                };
                const handleJobCompleted = (jobId, data) => {
                    if (jobId !== this._jobId)
                        return;
                    unsubscribe();
                    this.log.info("conversation complete");
                    if (this._recordingUrl !== undefined) {
                        this.log.info(`recording available at ${this._recordingUrl}`);
                    }
                    resolve({
                        output: data,
                        transcription: this._transcription.sort((a, b) => +a.startTime - +b.startTime),
                        recordingUrl: this._recordingUrl,
                    });
                };
                const handleJobRecordIdReady = (jobId, recordId) => {
                    if (jobId !== this._jobId)
                        return;
                    this._recordingUrl = this._application._getRecordingUrl(recordId);
                };
                const handleJobEvent = (jobId, jobEvent) => {
                    if (jobId !== this._jobId)
                        return;
                    this.emit("debugLog", jobEvent);
                    this._transcriptionParser.push(jobEvent);
                };
                this._application.on("_jobFailed", handleJobFailed);
                this._application.on("_jobCompleted", handleJobCompleted);
                this._application.on("_jobRecordIdReady", handleJobRecordIdReady);
                this._application.on("_jobEvent", handleJobEvent);
            });
        });
    }
}
exports.QueuedConversation = QueuedConversation;
