// @dasha.ai/sdk/dist/rest-api/generated/api/diagnostics-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiagnosticsApi = exports.DiagnosticsApiFactory = exports.DiagnosticsApiFp = exports.DiagnosticsApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * DiagnosticsApi - axios parameter creator
 * @export
 */
const DiagnosticsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerLimitGet: (options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/api/v1/diagnostics/customer/limit`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueueLengthGet: (options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/api/v1/diagnostics/customer/queue/length`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueuesLengthsGet: (options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/api/v1/diagnostics/customer/queues/lengths`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet: (groupId, applicationName, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'groupId' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet', 'groupId', groupId);
            // verify required parameter 'applicationName' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet', 'applicationName', applicationName);
            const localVarPath = `/api/v1/diagnostics/group/{groupId}/application/{applicationName}/limit`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)))
                .replace(`{${"applicationName"}}`, encodeURIComponent(String(applicationName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet: (groupId, applicationName, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'groupId' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet', 'groupId', groupId);
            // verify required parameter 'applicationName' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet', 'applicationName', applicationName);
            const localVarPath = `/api/v1/diagnostics/group/{groupId}/application/{applicationName}/queue/length`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)))
                .replace(`{${"applicationName"}}`, encodeURIComponent(String(applicationName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdLimitGet: (groupId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'groupId' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsGroupGroupIdLimitGet', 'groupId', groupId);
            const localVarPath = `/api/v1/diagnostics/group/{groupId}/limit`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdQueueLengthGet: (groupId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'groupId' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsGroupGroupIdQueueLengthGet', 'groupId', groupId);
            const localVarPath = `/api/v1/diagnostics/group/{groupId}/queue/length`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet: (instanceId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'instanceId' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/diagnostics/instance/{instanceId}/all/queue/length`
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitGet: (instanceId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'instanceId' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsInstanceInstanceIdLimitGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/diagnostics/instance/{instanceId}/limit`
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitsGet: (instanceId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'instanceId' is not null or undefined
            common_1.assertParamExists('apiV1DiagnosticsInstanceInstanceIdLimitsGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/diagnostics/instance/{instanceId}/limits`
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsLimitsGet: (options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/api/v1/diagnostics/limits`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} customerId
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet: (customerId, instanceId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'customerId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet', 'customerId', customerId);
            // verify required parameter 'instanceId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/management/diagnostics/customer/{customerId}/instance/{instanceId}/all/queue/length`
                .replace(`{${"customerId"}}`, encodeURIComponent(String(customerId)))
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} customerId
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet: (customerId, instanceId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'customerId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet', 'customerId', customerId);
            // verify required parameter 'instanceId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/management/diagnostics/customer/{customerId}/instance/{instanceId}/limits`
                .replace(`{${"customerId"}}`, encodeURIComponent(String(customerId)))
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdLimitGet: (customerId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'customerId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsCustomerCustomerIdLimitGet', 'customerId', customerId);
            const localVarPath = `/api/v1/management/diagnostics/customer/{customerId}/limit`
                .replace(`{${"customerId"}}`, encodeURIComponent(String(customerId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdLimitsGet: (customerId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'customerId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsCustomerCustomerIdLimitsGet', 'customerId', customerId);
            const localVarPath = `/api/v1/management/diagnostics/customer/{customerId}/limits`
                .replace(`{${"customerId"}}`, encodeURIComponent(String(customerId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdQueueLengthGet: (customerId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'customerId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsCustomerCustomerIdQueueLengthGet', 'customerId', customerId);
            const localVarPath = `/api/v1/management/diagnostics/customer/{customerId}/queue/length`
                .replace(`{${"customerId"}}`, encodeURIComponent(String(customerId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdQueuesLengthsGet: (customerId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'customerId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsCustomerCustomerIdQueuesLengthsGet', 'customerId', customerId);
            const localVarPath = `/api/v1/management/diagnostics/customer/{customerId}/queues/lengths`
                .replace(`{${"customerId"}}`, encodeURIComponent(String(customerId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet: (groupId, applicationName, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'groupId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet', 'groupId', groupId);
            // verify required parameter 'applicationName' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet', 'applicationName', applicationName);
            const localVarPath = `/api/v1/management/diagnostics/group/{groupId}/application/{applicationName}/limit`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)))
                .replace(`{${"applicationName"}}`, encodeURIComponent(String(applicationName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet: (groupId, applicationName, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'groupId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet', 'groupId', groupId);
            // verify required parameter 'applicationName' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet', 'applicationName', applicationName);
            const localVarPath = `/api/v1/management/diagnostics/group/{groupId}/application/{applicationName}/queue/length`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)))
                .replace(`{${"applicationName"}}`, encodeURIComponent(String(applicationName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdLimitGet: (groupId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'groupId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsGroupGroupIdLimitGet', 'groupId', groupId);
            const localVarPath = `/api/v1/management/diagnostics/group/{groupId}/limit`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdQueueLengthGet: (groupId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'groupId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsGroupGroupIdQueueLengthGet', 'groupId', groupId);
            const localVarPath = `/api/v1/management/diagnostics/group/{groupId}/queue/length`
                .replace(`{${"groupId"}}`, encodeURIComponent(String(groupId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsInstanceInstanceIdLimitGet: (instanceId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'instanceId' is not null or undefined
            common_1.assertParamExists('apiV1ManagementDiagnosticsInstanceInstanceIdLimitGet', 'instanceId', instanceId);
            const localVarPath = `/api/v1/management/diagnostics/instance/{instanceId}/limit`
                .replace(`{${"instanceId"}}`, encodeURIComponent(String(instanceId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsSchedulerGet: (options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/api/v1/management/diagnostics/scheduler`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.DiagnosticsApiAxiosParamCreator = DiagnosticsApiAxiosParamCreator;
/**
 * DiagnosticsApi - functional programming interface
 * @export
 */
const DiagnosticsApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.DiagnosticsApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerLimitGet(options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsCustomerLimitGet(options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueueLengthGet(options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsCustomerQueueLengthGet(options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueuesLengthsGet(options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsCustomerQueuesLengthsGet(options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsLimitsGet(options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1DiagnosticsLimitsGet(options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} customerId
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet(customerId, instanceId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet(customerId, instanceId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} customerId
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet(customerId, instanceId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet(customerId, instanceId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdLimitGet(customerId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsCustomerCustomerIdLimitGet(customerId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdLimitsGet(customerId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsCustomerCustomerIdLimitsGet(customerId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdQueueLengthGet(customerId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsCustomerCustomerIdQueueLengthGet(customerId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdQueuesLengthsGet(customerId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsCustomerCustomerIdQueuesLengthsGet(customerId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdLimitGet(groupId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsGroupGroupIdLimitGet(groupId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsGroupGroupIdQueueLengthGet(groupId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsInstanceInstanceIdLimitGet(instanceId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsSchedulerGet(options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementDiagnosticsSchedulerGet(options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.DiagnosticsApiFp = DiagnosticsApiFp;
/**
 * DiagnosticsApi - factory interface
 * @export
 */
const DiagnosticsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.DiagnosticsApiFp(configuration);
    return {
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerLimitGet(options) {
            return localVarFp.apiV1DiagnosticsCustomerLimitGet(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueueLengthGet(options) {
            return localVarFp.apiV1DiagnosticsCustomerQueueLengthGet(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsCustomerQueuesLengthsGet(options) {
            return localVarFp.apiV1DiagnosticsCustomerQueuesLengthsGet(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
            return localVarFp.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
            return localVarFp.apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options) {
            return localVarFp.apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
            return localVarFp.apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options) {
            return localVarFp.apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
            return localVarFp.apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options) {
            return localVarFp.apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1DiagnosticsLimitsGet(options) {
            return localVarFp.apiV1DiagnosticsLimitsGet(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} customerId
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet(customerId, instanceId, options) {
            return localVarFp.apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet(customerId, instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} customerId
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet(customerId, instanceId, options) {
            return localVarFp.apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet(customerId, instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdLimitGet(customerId, options) {
            return localVarFp.apiV1ManagementDiagnosticsCustomerCustomerIdLimitGet(customerId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdLimitsGet(customerId, options) {
            return localVarFp.apiV1ManagementDiagnosticsCustomerCustomerIdLimitsGet(customerId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdQueueLengthGet(customerId, options) {
            return localVarFp.apiV1ManagementDiagnosticsCustomerCustomerIdQueueLengthGet(customerId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} customerId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsCustomerCustomerIdQueuesLengthsGet(customerId, options) {
            return localVarFp.apiV1ManagementDiagnosticsCustomerCustomerIdQueuesLengthsGet(customerId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
            return localVarFp.apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {string} applicationName
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
            return localVarFp.apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdLimitGet(groupId, options) {
            return localVarFp.apiV1ManagementDiagnosticsGroupGroupIdLimitGet(groupId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} groupId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
            return localVarFp.apiV1ManagementDiagnosticsGroupGroupIdQueueLengthGet(groupId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} instanceId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
            return localVarFp.apiV1ManagementDiagnosticsInstanceInstanceIdLimitGet(instanceId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementDiagnosticsSchedulerGet(options) {
            return localVarFp.apiV1ManagementDiagnosticsSchedulerGet(options).then((request) => request(axios, basePath));
        },
    };
};
exports.DiagnosticsApiFactory = DiagnosticsApiFactory;
/**
 * DiagnosticsApi - object-oriented interface
 * @export
 * @class DiagnosticsApi
 * @extends {BaseAPI}
 */
class DiagnosticsApi extends base_1.BaseAPI {
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsCustomerLimitGet(options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsCustomerLimitGet(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsCustomerQueueLengthGet(options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsCustomerQueueLengthGet(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsCustomerQueuesLengthsGet(options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsCustomerQueuesLengthsGet(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {string} applicationName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {string} applicationName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsGroupGroupIdLimitGet(groupId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsGroupGroupIdQueueLengthGet(groupId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsInstanceInstanceIdAllQueueLengthGet(instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsInstanceInstanceIdLimitGet(instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsInstanceInstanceIdLimitsGet(instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1DiagnosticsLimitsGet(options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1DiagnosticsLimitsGet(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} customerId
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet(customerId, instanceId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdAllQueueLengthGet(customerId, instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} customerId
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet(customerId, instanceId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsCustomerCustomerIdInstanceInstanceIdLimitsGet(customerId, instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} customerId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsCustomerCustomerIdLimitGet(customerId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsCustomerCustomerIdLimitGet(customerId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} customerId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsCustomerCustomerIdLimitsGet(customerId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsCustomerCustomerIdLimitsGet(customerId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} customerId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsCustomerCustomerIdQueueLengthGet(customerId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsCustomerCustomerIdQueueLengthGet(customerId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} customerId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsCustomerCustomerIdQueuesLengthsGet(customerId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsCustomerCustomerIdQueuesLengthsGet(customerId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {string} applicationName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameLimitGet(groupId, applicationName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {string} applicationName
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsGroupGroupIdApplicationApplicationNameQueueLengthGet(groupId, applicationName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsGroupGroupIdLimitGet(groupId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsGroupGroupIdLimitGet(groupId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} groupId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsGroupGroupIdQueueLengthGet(groupId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsGroupGroupIdQueueLengthGet(groupId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} instanceId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsInstanceInstanceIdLimitGet(instanceId, options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsInstanceInstanceIdLimitGet(instanceId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DiagnosticsApi
     */
    apiV1ManagementDiagnosticsSchedulerGet(options) {
        return exports.DiagnosticsApiFp(this.configuration).apiV1ManagementDiagnosticsSchedulerGet(options).then((request) => request(this.axios, this.basePath));
    }
}
exports.DiagnosticsApi = DiagnosticsApi;
