// @dasha.ai/sdk/dist/rest-api/generated/api/register-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisterApi = exports.RegisterApiFactory = exports.RegisterApiFp = exports.RegisterApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * RegisterApi - axios parameter creator
 * @export
 */
const RegisterApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {CustomerRegisterRequestDTO} [customerRegisterRequestDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementRegisterPost: (customerRegisterRequestDTO, options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/api/v1/management/register`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json-patch+json';
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = common_1.serializeDataIfNeeded(customerRegisterRequestDTO, localVarRequestOptions, configuration);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @param {string} name
         * @param {number} [maxConcurrency]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1RegisterPost: (name, maxConcurrency, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'name' is not null or undefined
            common_1.assertParamExists('apiV1RegisterPost', 'name', name);
            const localVarPath = `/api/v1/register`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            if (maxConcurrency !== undefined) {
                localVarQueryParameter['maxConcurrency'] = maxConcurrency;
            }
            if (name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.RegisterApiAxiosParamCreator = RegisterApiAxiosParamCreator;
/**
 * RegisterApi - functional programming interface
 * @export
 */
const RegisterApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.RegisterApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @param {CustomerRegisterRequestDTO} [customerRegisterRequestDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementRegisterPost(customerRegisterRequestDTO, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1ManagementRegisterPost(customerRegisterRequestDTO, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @param {string} name
         * @param {number} [maxConcurrency]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1RegisterPost(name, maxConcurrency, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1RegisterPost(name, maxConcurrency, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.RegisterApiFp = RegisterApiFp;
/**
 * RegisterApi - factory interface
 * @export
 */
const RegisterApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.RegisterApiFp(configuration);
    return {
        /**
         *
         * @param {CustomerRegisterRequestDTO} [customerRegisterRequestDTO]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1ManagementRegisterPost(customerRegisterRequestDTO, options) {
            return localVarFp.apiV1ManagementRegisterPost(customerRegisterRequestDTO, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @param {string} name
         * @param {number} [maxConcurrency]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1RegisterPost(name, maxConcurrency, options) {
            return localVarFp.apiV1RegisterPost(name, maxConcurrency, options).then((request) => request(axios, basePath));
        },
    };
};
exports.RegisterApiFactory = RegisterApiFactory;
/**
 * RegisterApi - object-oriented interface
 * @export
 * @class RegisterApi
 * @extends {BaseAPI}
 */
class RegisterApi extends base_1.BaseAPI {
    /**
     *
     * @param {CustomerRegisterRequestDTO} [customerRegisterRequestDTO]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof RegisterApi
     */
    apiV1ManagementRegisterPost(customerRegisterRequestDTO, options) {
        return exports.RegisterApiFp(this.configuration).apiV1ManagementRegisterPost(customerRegisterRequestDTO, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @param {string} name
     * @param {number} [maxConcurrency]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof RegisterApi
     */
    apiV1RegisterPost(name, maxConcurrency, options) {
        return exports.RegisterApiFp(this.configuration).apiV1RegisterPost(name, maxConcurrency, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.RegisterApi = RegisterApi;
