// @dasha.ai/sdk/dist/rest-api/generated/api/session-log-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Dasha.AI Platform API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionLogApi = exports.SessionLogApiFactory = exports.SessionLogApiFp = exports.SessionLogApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * SessionLogApi - axios parameter creator
 * @export
 */
const SessionLogApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsDebugJobIdGet: (jobId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'jobId' is not null or undefined
            common_1.assertParamExists('apiV1LogsDebugJobIdGet', 'jobId', jobId);
            const localVarPath = `/api/v1/logs/debug/{jobId}`
                .replace(`{${"jobId"}}`, encodeURIComponent(String(jobId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication Bearer required
            yield common_1.setApiKeyToObject(localVarHeaderParameter, "Authorization", configuration);
            // authentication oauth2 required
            // oauth required
            yield common_1.setOAuthToObject(localVarHeaderParameter, "oauth2", ["platform_api"], configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter, options.query);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.SessionLogApiAxiosParamCreator = SessionLogApiAxiosParamCreator;
/**
 * SessionLogApi - functional programming interface
 * @export
 */
const SessionLogApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.SessionLogApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsDebugJobIdGet(jobId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.apiV1LogsDebugJobIdGet(jobId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.SessionLogApiFp = SessionLogApiFp;
/**
 * SessionLogApi - factory interface
 * @export
 */
const SessionLogApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.SessionLogApiFp(configuration);
    return {
        /**
         *
         * @param {string} jobId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        apiV1LogsDebugJobIdGet(jobId, options) {
            return localVarFp.apiV1LogsDebugJobIdGet(jobId, options).then((request) => request(axios, basePath));
        },
    };
};
exports.SessionLogApiFactory = SessionLogApiFactory;
/**
 * SessionLogApi - object-oriented interface
 * @export
 * @class SessionLogApi
 * @extends {BaseAPI}
 */
class SessionLogApi extends base_1.BaseAPI {
    /**
     *
     * @param {string} jobId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SessionLogApi
     */
    apiV1LogsDebugJobIdGet(jobId, options) {
        return exports.SessionLogApiFp(this.configuration).apiV1LogsDebugJobIdGet(jobId, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.SessionLogApi = SessionLogApi;
