// @dasha.ai/sdk/dist/rest-api/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TtsApi = exports.SipOutboundApi = exports.SipEndpointApi = exports.MiscApi = exports.GroupApi = exports.DiagnosticsApi = exports.CustomerApi = exports.models = void 0;
exports.models = require("./generated/models");
var customer_1 = require("./customer");
Object.defineProperty(exports, "CustomerApi", { enumerable: true, get: function () { return customer_1.CustomerApi; } });
var diagnostics_1 = require("./diagnostics");
Object.defineProperty(exports, "DiagnosticsApi", { enumerable: true, get: function () { return diagnostics_1.DiagnosticsApi; } });
var group_1 = require("./group");
Object.defineProperty(exports, "GroupApi", { enumerable: true, get: function () { return group_1.GroupApi; } });
var misc_1 = require("./misc");
Object.defineProperty(exports, "MiscApi", { enumerable: true, get: function () { return misc_1.MiscApi; } });
var sip_endpoint_1 = require("./sip-endpoint");
Object.defineProperty(exports, "SipEndpointApi", { enumerable: true, get: function () { return sip_endpoint_1.SipEndpointApi; } });
var sip_outbound_1 = require("./sip-outbound");
Object.defineProperty(exports, "SipOutboundApi", { enumerable: true, get: function () { return sip_outbound_1.SipOutboundApi; } });
var tts_1 = require("./tts");
Object.defineProperty(exports, "TtsApi", { enumerable: true, get: function () { return tts_1.TtsApi; } });
